packages/core/strapi/src/cli/commands/transfer/command.ts

```diff
@@ -44,14 +44,128 @@ const command = () => {
       .hook('preAction', validateExcludeOnly)
       .hook(
         'preAction',
-        ifOptions(
-          (opts) => !(opts.from || opts.to) || (opts.from && opts.to),
-          async () =>
+        async (thisCommand) => {
+          const opts = thisCommand.opts();
+          
+          // If neither --from nor --to are provided, prompt for transfer direction
+          if (!(opts.from || opts.to)) {
+            // Log environment variables status
+            const hasUrl = Boolean(process.env.STRAPI_TRANSFER_URL);
+            const hasToken = Boolean(process.env.STRAPI_TRANSFER_TOKEN);
+
+            console.info('ℹ️  Data transfer documentation: https://docs.strapi.io/dev-docs/data-management/transfer');
+
+            if (hasUrl && hasToken) {
+              console.info('ℹ️  Found transfer configuration in your environment:');
+              console.info('   → STRAPI_TRANSFER_URL environment variable value will be used as default URL');
+              console.info('   → STRAPI_TRANSFER_TOKEN environment variable value will be used as default token');
+            } else if (hasUrl || hasToken) {
+              if (hasUrl) {
+                console.info('ℹ️  Found partial transfer configuration:');
+                console.info('   → STRAPI_TRANSFER_URL environment variable value will be used as default URL');
+                console.info('   → STRAPI_TRANSFER_TOKEN environment variable not found ');
+              } else {
+                console.info('ℹ️  Found partial transfer configuration:');
+                console.info('   → STRAPI_TRANSFER_URL environment variable not found');
+                console.info('   → STRAPI_TRANSFER_TOKEN environment variable value will be used as default token');
+              }
```

Comment:
```suggestion
            if (hasUrl || hasToken) {
              console.info('ℹ️  Found transfer configuration in your environment:');
              if (hasUrl) {
                console.info('   → STRAPI_TRANSFER_URL environment variable value will be used as default URL');
              }
              if(hasToken) {
                console.info('   → STRAPI_TRANSFER_TOKEN environment variable value will be used as default token');
              }
```

I found this a little confusing/distracting when I was testing it. If I only set one of the two, I don't expect to see the other, and my brain doesn't want to read all that but is forced to to find out why there's something I didn't set, it felt like something went wrong 😆 For me, I'd prefer to just see the item that I set.

If we do want to keep it this way, maybe we could make the "found" items green or something and the not-found items yellow or something?